runtime/interpreter/decode.go

```diff
@@ -318,6 +319,58 @@ func (d *Decoder) decodeDictionary(v interface{}, path []string) (*DictionaryVal
 	keyCount := keys.Count()
 	entryCount := len(encodedEntries)
 
+	// In versions <= 2, the dictionary key string function
+	// was accidentally, temporarily changed without a version change.
+	//
+	// The key string format for address values is:
+	// prefix the address with 0x, encode in hex, and strip leading zeros.
+	//
+	// Temporarily and accidentally the format was:
+	// no 0x prefix, and encode and in full length hex.
+	//
+	// Detect this temporary format and correct it
+
+	var hasAddressValueKeyInWrongPre3Format bool
+
+	if d.version <= 2 {
+		for _, keyValue := range keys.Values {
+			keyAddressValue, ok := keyValue.(AddressValue)
+			if !ok {
+				continue
+			}
+
+			currentKeyString := keyAddressValue.KeyString()
+			wrongKeyString := hex.EncodeToString(keyAddressValue[:])
+
+			// Is the value for this key stored using the wrong key string format for address values?
```

Comment:
we might add an extra condition to return early if currentKeyString exists

Reply:
Good point, refactored in 7ceb1ad